integration/test/SetupHelpers.hs

```diff
@@ -11,6 +12,12 @@ import GHC.Stack
 import Testlib.Prelude
 import Testlib.ResourcePool (remoteDomains)
 
+-- | `n` should be 2 x `setFederationDomainConfigsUpdateFreq` in the config
+connectAllDomainsAndWaitToSync :: HasCallStack => Int -> [String] -> App ()
+connectAllDomainsAndWaitToSync n domains = do
+  sequence_ [Internal.createFedConn x (Internal.FedConn y "full_search") | x <- domains, y <- domains, x /= y]
+  liftIO $ threadDelay (n * 1000 * 1000) -- wait for federation status to be updated
```

Comment:
This is probably going to create flakes. Is there no way of observing the changes? E.g. with an internal endpoint.